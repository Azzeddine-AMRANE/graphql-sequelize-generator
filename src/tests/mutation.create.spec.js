const request = require('supertest')
const {
  migrateDatabase,
  seedDatabase,
  deleteTables
} = require('./testDatabase.js')
const express = require('express')
const { graphqlExpressMiddleware } = require('./schema')
const http = require('spdy')

let app = null
let server = null

var options = {
  spdy: {
    plain: true
  }
}

/**
 * Starting the tests
 */
describe('Test the API queries', () => {
  beforeAll(async () => {
    app = express()
    app.use('/graphql', graphqlExpressMiddleware)
    server = await new Promise((resolve, reject) => {
      const newServer = http
        .createServer(options, app)
        .listen(process.env.PORT || 8080, () => {
          resolve(newServer)
        })
    })
  })

  afterAll(() =>
    Promise.all([
      new Promise((resolve, reject) => server.close(() => resolve()))
    ])
  )

  beforeEach(async () => {
    try {
      await migrateDatabase()
      await seedDatabase()
    } catch (e) {
      console.log('Critical error during the database migration', e.message, e)
      throw e
    }
  })

  afterEach(async () => {
    await deleteTables()
  })

  it('Check that you can query a model and an association', async () => {
    const responseMutation = await request(server)
      .post('/graphql')
      .set('userid', 1)
      .send({
        query: `mutation companyCreate($company: companyInput) {
              company : companyCreate(company: $company) {
                id
                name
                __typename
              }
            }`,
        variables: {
          company: {
            name: 'test',
            companyTypeId: 1
          }
        },
        operationName: 'companyCreate'
      })
    expect(responseMutation.body.data.company).toMatchSnapshot()

    const response = await request(server)
      .get(
        `/graphql?query=
          query getCompanies {
            companies: company {
              id
              name
              users {
                id
                name
                department {
                  id
                }
              }
            }
          }
          &operationName=getCompanies`
      )
      .set('userId', 1)
    const companies = response.body.data.companies
    expect(companies).toMatchSnapshot(
      'All companies with users and their department'
    )
  })
})
