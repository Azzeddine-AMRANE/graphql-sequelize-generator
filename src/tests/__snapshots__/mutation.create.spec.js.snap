// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the create mutation Check if you cannot duplicate entity according to attributes 1`] = `
Object {
  "userCreate": Object {
    "id": 25001,
    "name": "new user",
  },
}
`;

exports[`Test the create mutation Check that you can create a new instance of one entity 1`] = `
Array [
  "listGlobalBefore",
  "listBefore",
  "listAfter",
  "createGlobalBefore",
  "createBefore",
  "createAfter",
  "listGlobalBefore",
  "listBefore",
  "listAfter",
]
`;

exports[`Test the create mutation Check that you can create a new instance of one entity: Created user 1`] = `
Object {
  "__typename": "user",
  "id": 25001,
  "name": "new user",
}
`;

exports[`Test the create mutation Check that you can create a new instance of one entity: The new user 1`] = `Array []`;

exports[`Test the create mutation Check that you can create a new instance of one entity: User 12501 does not exists yet. 1`] = `Array []`;
